core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -878,19 +885,28 @@ private[kafka] class Processor(val id: Int,
 
 class ConnectionQuotas(val defaultMax: Int, overrideQuotas: Map[String, Int]) {
 
-  private val overrides = overrideQuotas.map { case (host, count) => (InetAddress.getByName(host), count) }
+  private var defaultMaxConnectionsPerIp = defaultMax
+  private var maxConnectionsPerIpOverrides = overrideQuotas.map { case (host, count) => (InetAddress.getByName(host), count) }
```

Comment:
These two vars should be marked `volatile` since they are being updated without synchronization.